docs/src/js/classes/Session.md

```diff
@@ -0,0 +1,84 @@
+[**@lancedb/lancedb**](../README.md) • **Docs**
+
+***
+
+[@lancedb/lancedb](../globals.md) / Session
+
+# Class: Session
+
+A session for managing caches and object stores across LanceDB operations.
+
+Sessions allow you to configure cache sizes for index and metadata caches,
+which can significantly impact performance for large datasets.
```

Comment:
I would maybe mention here how they allow you to share cache across different lancedb connections. That's the main way they improve performance right?

Had a similar thought on the deprecation messages (repeated in a few places). We end with "are more efficient" but could say something like "they enable cache reuse across connections".